src/sql/src/plan/lowering.rs

```diff
@@ -1137,6 +1157,38 @@ impl HirScalarExpr {
             ScalarType,
         ) -> Result<(MirScalarExpr, ColumnType), PlanError>,
     {
+        // Example MIRs for a window function (specifically, a window aggregation):
+        //
+        // CREATE TABLE t7(x INT, y INT);
+        //
+        // explain decorrelated plan for select sum(x*y) over (partition by x+y order by x-y, x/y) from t7;
+        //
+        // Decorrelated Plan
+        // Project (#3)
+        //   Map (#2)
+        //     Project (#3..=#5)
+        //       Map (record_get[0](record_get[1](#2)), record_get[1](record_get[1](#2)), record_get[0](#2))
+        //         FlatMap unnest_list(#1)
+        //           Reduce group_by=[#2] aggregates=[window_agg[sum order_by=[#0 asc nulls_last, #1 asc nulls_last]](row(row(row(#0, #1), (#0 * #1)), (#0 - #1), (#0 / #1)))]
```

Comment:
Can you spell out in this comment what the various levels of `row()` depth are meant to represent.

I am always confused when I see the `row(row(row()))` and the `record_get` chains in the `Map` above.

Some related questions:

1. Is the `row(#0, #1)` needed for a window aggregation?

Reply:
Yes, I've added an extended comment about this now.

> Is the row(#0, #1) needed for a window aggregation?

Yes, this is the original input row. This has to survive the MIR aggregation of the window aggregate, because columns from the original input row might be used outside the window function call. E.g. if the user wrote
```
select x, y, sum(x*y) over (partition by x+y order by x-y, x/y) from t7;
```
then we need both `x` and `y,` in addition to the result of the window aggregate. Note that in some cases we could omit some columns of the original row if we made `ProjectionPushdown` smarter, which is tracked by https://github.com/MaterializeInc/materialize/issues/17522.

Reply:
> This has to survive the MIR aggregation of the window aggregate, because columns from the original input row might be used outside the window function call.

Yes I forgot about that part.